internal/raft/append_entries.go

```diff
@@ -6,6 +6,52 @@ import "github.com/tomarrell/lbadd/internal/raft/message"
 // to the follower node. This function generates the response to be sent to the leader node.
 // This is the response to the contact by the leader to assert it's leadership.
 func AppendEntriesResponse(node *Node, req *message.AppendEntriesRequest) *message.AppendEntriesResponse {
+	success := true
+	leaderTerm := req.GetTerm()
+	nodePersistentState := node.PersistentState
+	nodeTerm := nodePersistentState.CurrentTerm
+	// return false if term < currentTerm
+	if nodeTerm > leaderTerm {
+		success = false
+	} else if req.GetPrevLogIndex() > node.VolatileState.CommitIndex {
+		// return false if msg Log Index is greater than node commit Index
+		success = false
+	} else if nodePersistentState.Log[req.PrevLogIndex].Term != req.GetPrevLogTerm() {
+		// return false if term of msg at PrevLogIndex doesn't match prev Log Term stored by Leader
+		success = false
+	}
+
+	if !success {
+		return &message.AppendEntriesResponse{
+			Term:    nodeTerm,
+			Success: success,
+		}
+	}
```

Comment:
Why not combine this?

Reply:
Every Check has to be passed before log can append to node PersistentState Log. How you want me to combine this ?

Reply:
why not immediately return?

Reply:
In this case also it will immediately return after any case is true ?

Reply:
You point being?

Reply:
You could also reduce it to one OR-ed expression
or an inline value for success within the return

Reply:
okay i will update it with one OR-ed expression